pkg/secrets/k8s_secrets_storage/provide_conjur_secrets_test.go

```diff
@@ -372,6 +375,7 @@ func TestProvide(t *testing.T) {
 		for secretName, secretData := range tc.k8sSecrets {
 			mocks.kubeClient.AddSecret(secretName, secretData)
 		}
+		prevSecretsChecksums = map[string]utils.Checksum{}
```

Comment:
Doesn't look like this is used anywhere

Reply:
This is needed for TestProvide() to zero out prevSecretsChecksums so that it writes values in this test.

Reply:
Oooh.... I'm glad @szh asked this question!!!  I don't think we should be using a global variable for `prevSecretsChecksums` here, or for the Push-to-File provider either. The issue that we might face with using global variables is that if our UT test cases are ever run in parallel, then one test case could potentially stomp on another test case's values in the global `prevSecretsChecksums`.

I believe the `prevSecretsChecksums` should be moved to the `K8sProvider` struct. Each UT test case should be creating its own `K8sProvider` to test with, so each one should be created with its own `prevSecretsChecksums`. Thinking about this a bit more, this would also help in the future if we should ever need to run multiple `K8sProviders` in parallel.

We need to fix this for the Push-to-File provider as well. I don't know if you want to tackle that here or separately.